opensrp-app/src/main/java/org/smartregister/repository/P2PSenderTransferDao.java

```diff
@@ -25,40 +27,73 @@
 
 public class P2PSenderTransferDao extends BaseP2PTransferDao implements SenderTransferDao {
 
+    private static final String SEPARATOR = "-";
+
+    private P2POptions p2POptions;
+
+    public P2PSenderTransferDao() {
+        this(null);
+    }
+
+    public P2PSenderTransferDao(P2POptions p2POptions) {
+        super();
+        this.p2POptions = p2POptions;
+    }
+
     @Nullable
     @Override
     public TreeSet<DataType> getDataTypes() {
-        return (TreeSet<DataType>) dataTypes.clone();
+        TreeSet<DataType> dataTypeTreeSet = new TreeSet<>();
+
+        if (locationFilterEnabled()) {
+            for (String location : p2POptions.getLocationsFilter()) {
+                for (DataType dataType : dataTypes) {
+                    dataTypeTreeSet.add(new DataType(dataType.getName() + SEPARATOR + location, dataType.getType(), dataTypeTreeSet.size()));
```

Comment:
Set the `DataType#position` in the third param as the loop position. This is used to order the datatypes so that the syncing is done in the correct order.

I think you should also change the loops i.e the dataType loop should be the outer loop so that the order is 
1. client-{locationId1}
2. client-{locationId2}
3. client-{locationId3}
4. event-{locationId1}
5. event-{locationId2} ...

Reply:
Not sure wanted for it order sync data per locations, so that if the user checks a particular location chances are that data is synced or its not. Ordering the data type maybe confusing when user opens an location with some data synced while others is not synched